index.js

```diff
@@ -6,9 +6,18 @@ const { GitHub, context } = require('@actions/github')
 const main = async () => {
   const token = core.getInput('github-token')
   const number = core.getInput('number')
-
   const octokit = new GitHub(token)
 
+  // Check for existing approvals before approving a second time
+  const { data: reviews } = await octokit.pulls.listReviews({
+    ...context.repo,
+    pull_number: number
+  })
+  const user = await octokit.users.getAuthenticated()
+  const approved = reviews.some(review => review.state === 'APPROVED' && review.user.login === user.login)
```

Comment:
This bit won't work - `users.getAuthenticated` returns the authed user based on the token, in this case that's a GitHub Actions token, not the user. You'd want something like:

```suggestion
  const approved = reviews.some(review => review.state === 'APPROVED' && review.user.login === context.payload.sender.login)
```

`context.payload.sender` is a generic object that represents the actor that triggered the event. Depending on it's in an action, that may be the user who added a label or the user who opened a PR.

Reply:
Just to clarify, it's the @github-actions user that does the approving:

<img width="508" alt="Screen Shot 2020-09-24 at 8 52 15 PM" src="https://user-images.githubusercontent.com/2289/94224780-265fe100-fea8-11ea-9b93-e76c675c9428.png">

With that in mind, does your comment above still hold true?

Reply:
you don't know where the token is coming from. The token can be set from `${{secrets.GITHUB_TOKEN}}` and then it would be the `@github-actions[bot]`. But I could also pass `${{secrets.GR2M_PAT}}` and then it would by my own user account. If it's `${{secrets.GITHUB_TOKEN}}`, then the `octokit.users.getAuthenticated()` won't work, because it only works for OAuth tokens, not installation access tokens (which is what `GITHUB_TOKEN` is). For situations like this, I wish there was a [`GET /actor`](https://github.com/gr2m/github-api-wishlist/tree/master/wishlist/get-actor) endpoint ... 🤷🏼 

What I would recommend instead is to use GraphQL. You have two options.

1. Two queries

   Get the authenticated actor's login

   ```graphql
   {
     viewer {
       login
     }
   }
   ```

   Then use the login as query variable. The `$id` variable must be set to the pull requests `node_id` property  (`context.payload.pull_request.node_id`)

   ```graphql
   query($id: ID!, $login: String!) {
     node(id: $id) {
       ... on PullRequest {
         reviews(states:[APPROVED],author:$login,first:1) {
           totalCount
         }
       }
     }
   }
   ```

1. One query

   ```graphql
   query($id: ID!) {
     node(id: $id) {
       ... on PullRequest {
         reviews(states:[APPROVED],first:100) {
           totalCount
           nodes {
             viewerDidAuthor
           }
         }
       }
     }
   }
   ```

   Then iterate through the reviews and see if `viewerDidAuthor` is true for any result. You get up to 100 first results, but I guess that should be fine.